tests/helixpublishwitharcade.proj

```diff
@@ -29,57 +30,77 @@
 
     <WaitForWorkItemCompletion>true</WaitForWorkItemCompletion>
     <CoreRootDirectory>$(TestWorkingDir)\Tests\Core_Root</CoreRootDirectory>
-  </PropertyGroup>
-
-  <PropertyGroup Condition=" '$(TargetsWindows)' == 'true' ">
-    <HelixPythonPath>%HELIX_PYTHONPATH%</HelixPythonPath>
-    <RunnerScript>%HELIX_CORRELATION_PAYLOAD%\runtest_helix.py</RunnerScript>
-  </PropertyGroup>
-
-  <PropertyGroup Condition=" '$(TargetsWindows)' != 'true' ">
-    <HelixPythonPath>$HELIX_PYTHONPATH</HelixPythonPath>
-    <RunnerScript>$HELIX_CORRELATION_PAYLOAD/runtest_helix.py</RunnerScript>
+    <TestEnvFileExtension Condition=" '$(TargetsWindows)' == 'true' ">.cmd</TestEnvFileExtension>
+    <TestEnvFileExtension Condition=" '$(TargetsWindows)' != 'true' ">.sh</TestEnvFileExtension>
   </PropertyGroup>
 
   <ItemGroup>
     <HelixCorrelationPayload Include="$(CoreRootDirectory)" />
     <XUnitWrapperDlls Include="$(TestWorkingDir)\**\*.XUnitWrapper.dll" />
+    <Scenarios Include="$(_Scenarios.Split(','))" />
   </ItemGroup>
 
-  <Target Name="CopyRunnerScriptToCoreRootDirectory" BeforeTargets="SubmitTestsToHelix">
-    <Copy SourceFiles="$(MSBuildThisFileDirectory)\runtest_helix.py" DestinationFolder="$(CoreRootDirectory)" />
+  <Target Name="CreateTestEnvFiles">
+    <ItemGroup>
+      <_XUnitWrapperDirectoryWithScenario Include="@(XUnitWrapperDlls->'%(RootDir)%(Directory)')">
+        <Scenario>%(Scenarios.Identity)</Scenario>
+      </_XUnitWrapperDirectoryWithScenario>
+
+      <_ProjectsToBuild Include=".\testenvironment.proj">
+        <Properties>Scenario=%(_XUnitWrapperDirectoryWithScenario.Scenario);TestEnvFileName=%(Identity)\SetStressModes_%(Scenario)$(TestEnvFileExtension);TargetsWindows=$(TargetsWindows)</Properties>
+      </_ProjectsToBuild>
+    </ItemGroup>
+
+    <MSBuild Projects="@(_ProjectsToBuild)" Targets="CreateTestEnvFile" BuildInParallel="false" StopOnFirstFailure="true" UnloadProjectsOnCompletion="true" />
+
+    <ItemGroup>
+      <_ProjectsToBuild Remove="@(_ProjectsToBuild)" />
```

Comment:
I think it would be better to use a separate ItemGroup (`_TestEnvProjects`?) so that you don't need to clear it and worry about it being reused above. It gets confusing when the same itemgroup from the global namespace means different things in different targets.

Reply:
Agree, I will rename the items. Thanks for the suggestion!